### DISCUSSION TITLE
Node switchover in the cluster setup is not working

### DISCUSSION THREAD

**Discussion by mahesh-av-qp** [rabbitmq/rabbitmq-java-client#1118] [General]
### Describe the bug

I have a cluster setup with 3 nodes [host1, host2 & host3] and I have provided the address as follows
` connection = getConnectionFactory().newConnection(getAddresses());`
...
```
private Address[] getAddresses() {
        return Address.parseAddresses("host1:xxxx, host2:xxxx, host3:xxxx");
    }
```
Initially, my application was connected to host1.
When host1 goes down for some reason, queue leader switchover works fine and the rabbitMQ cluster continues to work fine as expected but my application keeps pushing the messages to host1 and fails.

### Reproduction steps

1. setup a  cluster with 3 nodes
2. kill the node that the app is connected to
3. continue pushing messages from the client

### Expected behavior

The client should be aware of the cluster and if the cluster is up, it should continue pushing the message to the nodes that are up & running.

Am I missing something?

**Comment by michaelklishin** [comment]
* Client libraries do not detect node failures instantly, not in all cases anyway. See [Heartbeats](https://rabbitmq.com/heartbeats.html)
* Connection recovery happens after a delay, by design, and you must opt-in (or roll your own recovery strategy, although that's usually not necessary). See [Automatic Recovery from Network Failures](https://rabbitmq.com/api-guide.html#recovery)
* Your application must use [publisher confirms](https://rabbitmq.com/publishers.html). When a node goes down, it will takes some time for the library to reconnect to another reachable node from the list. Making sure that all pending messages are then resent is your application's responsibility, the docs cover several common approaches that use publisher confirms

**Comment by michaelklishin** [comment]
This client is not aware of cluster topology and will simply retry all provided endpoints until one of the succeeds. This is the case for many reasons, including historical ones, but most importantly, the protocol does not assume that clients perform any kind of cluster topology discovery. They can connect to any node and perform any operations.

In contrast, RabbitMQ Streams Java client is topology aware, the protocol allows it to benefit from connection parallelism, but [it comes with a price tag in complexity](https://github.com/rabbitmq/rabbitmq-stream-java-client/pull/420) which can be a deal breaker for beginners.
